### DISCUSSION TITLE
How to analyze audio files post-recording?

### DISCUSSION THREAD

**Discussion by Vollpflock** [tphakala/birdnet-go#25] [Q&A]
Just a quick, simple question from a dummy.

You mentioned that you've launched BirdNET-Go, among other things, to implement some options that allow BirdNET-Go to function as a battery-powered remote station.

I have a much simpler question and need to take a step back:

Could you briefly explain how to analyze audio files post-recording? It seems that BirdNET-Analyzer isn't designed for this task (I do not believe that BirdNET-Analyzer was designed to display spectrograms or play back sounds.)

I'm asking because I have thousands of hours of audio material that needs to be analyzed.

**Comment by tphakala** [comment]
At the moment post processing works by running "birdnet-go file soundfile.wav" or "birdnet-go directory soundfiledir" but it does not produce any spectrograms or audio clips, it will produce similar table and cvs files as BirdNET-Analyzer does. File analysis implementation on BirdNET-Go is not very good at the moment, process works by reading whole wav file into memory first and then running it through BirdNET model, this works poorly with large wav files.

At the moment my priority is to get BirdNET-Go web UI in good enough shape to release it, I will improve file processing in future.

**Reply by Vollpflock** [reply]
Sounds great! I guess this would be much easier to implement than real-time remote analysis, as it's far more power-efficient to record first and analyze later.
